Lib/fontTools/designspaceLib/__init__.py

```diff
@@ -160,45 +160,51 @@ def __init__(self):
 class RuleDescriptor(SimpleDescriptor):
     """<!-- optional: list of substitution rules -->
     <rules>
-        <rule name="vertical.bars" enabled="true">
+        <rule name="vertical.bars">
+            <conditionset>
+                <condition minimum="250.000000" maximum="750.000000" name="weight"/>
+                <condition minimum="100" name="width"/>
+                <condition minimum="10" maximum="40" name="optical"/>
+            </conditionset>
             <sub name="cent" byname="cent.alt"/>
             <sub name="dollar" byname="dollar.alt"/>
-            <condition tag="wght" minimum ="250.000000" maximum ="750.000000"/>
-            <condition tag="wdth" minimum ="100"/>
-            <condition tag="opsz" minimum="10" maximum="40"/>
         </rule>
     </rules>
-
-    Discussion:
-    use axis names rather than tags - then we can evaluate the rule without having to look up the axes.
-    remove the subs from the rule.
-    remove 'enabled' attr form rule
     """
-    _attrs = ['name', 'conditions', 'subs']   # what do we need here
+    _attrs = ['name', 'conditionSets', 'subs']   # what do we need here
     def __init__(self):
         self.name = None
-        self.conditions = []    # list of dict(tag='aaaa', minimum=0, maximum=1000)
-        self.subs = []          # list of substitutions stored as tuples of glyphnames ("a", "a.alt")
+        self.conditionSets = []  # list of list of dict(name='aaaa', minimum=0, maximum=1000)
+        self.subs = []  # list of substitutions stored as tuples of glyphnames ("a", "a.alt")
+
 
 def evaluateRule(rule, location):
-    """ Test if rule is True at location.maximum
-        If a condition has no minimum, check for < maximum.
-        If a condition has no maximum, check for > minimum.
-     """
-    for cd in rule.conditions:
-        if not cd['name'] in location:
+    """ Return True if any of the rule's conditionsets matches the
+    given location.
+    """
+    return any(evaluateConditions(c, location) for c in rule.conditionSets)
+
+
+def evaluateConditions(conditions, location):
+    """ Return True if all the conditions matches the given location.
+    If a condition has no minimum, check for < maximum.
+    If a condition has no maximum, check for > minimum.
+    """
+    for cd in conditions:
+        if cd['name'] not in location:
             continue
```

Comment:
shouldn't this case -- when the requested location does not specify a value for an axis used inside a condition -- be documented? Is it like the location does not represent a single point, but a subspace (rectangle, box, etc.)?

Reply:
A location that does not have a value for a specific axis should be assumed to have the default value for that axis. This seemed easier than to assume all locations always have all dimensions.

Reply:
> should be assumed to have the default value for that axis

in that case we would need access to the axes in this evaluateRule function.
The current code would return True if location does not have explicit values for all the condition axes